### PR TITLE
Enable weighted fitting for both single and multi eval tree

### PR DESCRIPTION
**Pull request by hwpang**

### CODE REVIEW

File: pysidt/sidt.py [hunk 1]
```diff
@@ -855,7 +862,7 @@ def estimate_uncertainty(self):
 
         if A.shape[1] != 1:
             node_uncertainties = (
-                np.diag(np.linalg.pinv((A.T @ A).toarray()))
+                np.diag(np.linalg.pinv((A.T @ W @ A).toarray()))
```
Comment (mjohnson541) [inline pysidt/sidt.py:913]: I think this W matrix needs normalized to avoid affecting the uncertainties magnitude.

**Review by mjohnson541 — COMMENTED** [review]
Set W matrix in setup_data?

**Comment by hwpang** [comment]
@mjohnson541 Thanks, done

**Comment by hwpang** [comment]
@mjohnson541 I have rebased, please review and merge!

**Review by mjohnson541 — APPROVED** [review]

**Pull request merged**